error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:13:11
   |
5  | #[stream(item = Option<i32>)]
   | ----------------------------- arguments to this enum variant are incorrect
...
13 |     yield val; //~ ERROR mismatched types
   |           ^^^ expected enum `Option`, found integer
   |
   = note: expected enum `Option<_>`
              found type `{integer}`
note: tuple variant defined here
  --> $RUST/core/src/task/poll.rs
   |
   |     Ready(#[stable(feature = "futures_api", since = "1.36.0")] T),
   |     ^^^^^
help: try wrapping the expression in `Some`
   |
13 |     yield Some(val); //~ ERROR mismatched types
   |           +++++   +

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:32:11
   |
27 | #[stream(item = (i32, i32))]
   | ---------------------------- arguments to this enum variant are incorrect
...
32 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`
note: tuple variant defined here
  --> $RUST/core/src/task/poll.rs
   |
   |     Ready(#[stable(feature = "futures_api", since = "1.36.0")] T),
   |     ^^^^^

error[E0271]: type mismatch resolving `<impl Stream<Item = {integer}> as Stream>::Item == (i32, i32)`
  --> tests/ui/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `(i32, i32)`

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:24:11
   |
16 | #[stream(item = Option<i32>, boxed)]
   | ------------------------------------ arguments to this enum variant are incorrect
...
24 |     yield val; //~ ERROR mismatched types
   |           ^^^ expected enum `Option`, found integer
   |
   = note: expected enum `Option<_>`
              found type `{integer}`
note: tuple variant defined here
  --> $RUST/core/src/task/poll.rs
   |
   |     Ready(#[stable(feature = "futures_api", since = "1.36.0")] T),
   |     ^^^^^
help: try wrapping the expression in `Some`
   |
24 |     yield Some(val); //~ ERROR mismatched types
   |           +++++   +

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:40:11
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ----------------------------------- arguments to this enum variant are incorrect
...
40 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`
note: tuple variant defined here
  --> $RUST/core/src/task/poll.rs
   |
   |     Ready(#[stable(feature = "futures_api", since = "1.36.0")] T),
   |     ^^^^^

error[E0271]: type mismatch resolving `<impl Stream<Item = {integer}> as Stream>::Item == (i32, i32)`
  --> tests/ui/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found integer
   |
   = note: expected tuple `(i32, i32)`
               found type `{integer}`
   = note: required for the cast from `impl Stream<Item = {integer}>` to the object type `dyn Stream<Item = (i32, i32)> + Send`
   = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)
