error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:13:11
   |
13 |     yield val; //~ ERROR mismatched types
   |           ^^^ expected enum `Option`, found integer
   |
   = note: expected enum `Option<_>`
              found type `{integer}`
help: try wrapping the expression in `futures_async_stream::__private::Some`
   |
13 |     yield futures_async_stream::__private::Some(val); //~ ERROR mismatched types
   |           ++++++++++++++++++++++++++++++++++++++   +

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:32:11
   |
32 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`

error[E0271]: type mismatch resolving `<impl Stream<Item = {integer}> as Stream>::Item == (i32, i32)`
  --> tests/ui/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `(i32, i32)`

error[E0271]: type mismatch resolving `<impl Stream<Item = {integer}> as Stream>::Item == (i32, i32)`
  --> tests/ui/bad-item-type.rs:28:18
   |
28 |   async fn tuple() {
   |  __________________^
29 | |     if false {
30 | |         yield 3;
31 | |     }
32 | |     yield (1, 2) //~ ERROR mismatched types
33 | | }
   | |_^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `(i32, i32)`

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:24:11
   |
24 |     yield val; //~ ERROR mismatched types
   |           ^^^ expected enum `Option`, found integer
   |
   = note: expected enum `Option<_>`
              found type `{integer}`
help: try wrapping the expression in `futures_async_stream::__private::Some`
   |
24 |     yield futures_async_stream::__private::Some(val); //~ ERROR mismatched types
   |           ++++++++++++++++++++++++++++++++++++++   +

error[E0308]: mismatched types
  --> tests/ui/bad-item-type.rs:40:11
   |
40 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`

error[E0271]: type mismatch resolving `<impl Stream<Item = {integer}> as Stream>::Item == (i32, i32)`
  --> tests/ui/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found integer
   |
   = note: expected tuple `(i32, i32)`
               found type `{integer}`
   = note: required for the cast to the object type `dyn Stream<Item = (i32, i32)> + Send`
   = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)
