error[E0425]: cannot find value `__arg0` in this scope
 --> tests/ui/def-site.rs:9:5
  |
9 |     __arg0; //~ ERROR E0425
  |     ^^^^^^ not found in this scope

error[E0425]: cannot find value `__arg0` in this scope
  --> tests/ui/def-site.rs:14:5
   |
14 |     __arg0; //~ ERROR E0425
   |     ^^^^^^ not found in this scope

error[E0425]: cannot find value `__task_context` in this scope
  --> tests/ui/def-site.rs:19:5
   |
19 |     __task_context; //~ ERROR E0425
   |     ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__task_context` in this scope
  --> tests/ui/def-site.rs:25:5
   |
25 |     __task_context; //~ ERROR E0425
   |     ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__pinned` in this scope
  --> tests/ui/def-site.rs:35:10
   |
35 |         &__pinned; //~ ERROR E0425
   |          ^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__pinned` in this scope
  --> tests/ui/def-site.rs:42:10
   |
42 |         &__pinned; //~ ERROR E0425
   |          ^^^^^^^^ not found in this scope
