error[E0308]: mismatched types
  --> tests/ui/forget-semicolon.rs:10:5
   |
10 |     Some(()) //~ ERROR mismatched types
   |     ^^^^^^^^ expected `()`, found `Option<()>`
   |
   = note: expected unit type `()`
                   found enum `Option<()>`

error[E0308]: mismatched types
  --> tests/ui/forget-semicolon.rs:16:5
   |
16 |     Some(()) //~ ERROR mismatched types
   |     ^^^^^^^^ expected `()`, found `Option<()>`
   |
   = note: expected unit type `()`
                   found enum `Option<()>`
