error: async stream must be declared as async
 --> $DIR/invalid.rs:7:11
  |
7 |     const fn constness() {} //~ ERROR async stream must be declared as async
  |           ^^

error: async stream must be declared as async
  --> $DIR/invalid.rs:10:5
   |
10 |     fn variadic(_: ...) {} //~ ERROR only foreign functions are allowed to be C-variadic
   |     ^^

error: async stream must be declared as async
  --> $DIR/invalid.rs:13:5
   |
13 |     fn asyncness() {} //~ ERROR async stream must be declared as async
   |     ^^

error: async stream must return the unit type
  --> $DIR/invalid.rs:16:26
   |
16 |     async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                          ^^^

error: async stream may not be const
  --> $DIR/invalid.rs:22:5
   |
22 |     const async unsafe extern "C" fn f() {} //~ ERROR async stream may not be const
   |     ^^^^^

error: duplicate #[stream] attribute
  --> $DIR/invalid.rs:32:5
   |
32 |     #[stream(item = ())] //~ ERROR duplicate #[stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> $DIR/invalid.rs:36:5
   |
36 |     #[try_stream(ok = (), error = ())] //~ ERROR duplicate #[try_stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
  --> $DIR/invalid.rs:40:5
   |
40 |     #[try_stream(ok = (), error = ())] //~ ERROR may not be used at the same time
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: duplicate #[stream] attribute
  --> $DIR/invalid.rs:46:13
   |
46 |             #[stream] //~ ERROR duplicate #[stream] attribute
   |             ^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> $DIR/invalid.rs:54:13
   |
54 |             #[try_stream] //~ ERROR duplicate #[try_stream] attribute
   |             ^^^^^^^^^^^^^

error: duplicate #[for_await] attribute
  --> $DIR/invalid.rs:61:9
   |
61 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
   |         ^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
  --> $DIR/invalid.rs:68:13
   |
68 |             #[try_stream] //~ ERROR may not be used at the same time
   |             ^^^^^^^^^^^^^

error: duplicate #[stream] attribute
  --> $DIR/invalid.rs:77:13
   |
77 |             #[stream] //~ ERROR duplicate #[stream] attribute
   |             ^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> $DIR/invalid.rs:86:13
   |
86 |             #[try_stream] //~ ERROR duplicate #[try_stream] attribute
   |             ^^^^^^^^^^^^^

error: duplicate #[for_await] attribute
  --> $DIR/invalid.rs:94:9
   |
94 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
   |         ^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
   --> $DIR/invalid.rs:102:13
    |
102 |             #[try_stream] //~ ERROR may not be used at the same time
    |             ^^^^^^^^^^^^^

error: #[stream] attribute may only be used on async functions or async blocks
   --> $DIR/invalid.rs:111:5
    |
111 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: #[stream] attribute may only be used on async functions or async blocks
   --> $DIR/invalid.rs:114:5
    |
114 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: #[stream] attribute may only be used on async functions or async blocks
   --> $DIR/invalid.rs:117:5
    |
117 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)
