error: async stream must be declared as async
 --> tests/ui/invalid.rs:9:11
  |
9 |     const fn constness() {} //~ ERROR async stream must be declared as async
  |           ^^

error: async stream must be declared as async
  --> tests/ui/invalid.rs:12:5
   |
12 |     fn variadic(_: ...) {} //~ ERROR only foreign functions are allowed to be C-variadic
   |     ^^

error: async stream must be declared as async
  --> tests/ui/invalid.rs:15:5
   |
15 |     fn asyncness() {} //~ ERROR async stream must be declared as async
   |     ^^

error: async stream must return the unit type
  --> tests/ui/invalid.rs:18:26
   |
18 |     async fn output() -> i32 {} //~ ERROR async stream must return the unit type
   |                          ^^^

error: async stream may not be const
  --> tests/ui/invalid.rs:24:5
   |
24 |     const async unsafe extern "C" fn f() {} //~ ERROR async stream may not be const
   |     ^^^^^

error: duplicate #[stream] attribute
  --> tests/ui/invalid.rs:34:5
   |
34 |     #[stream(item = ())] //~ ERROR duplicate #[stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> tests/ui/invalid.rs:38:5
   |
38 |     #[try_stream(ok = (), error = ())] //~ ERROR duplicate #[try_stream] attribute
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
  --> tests/ui/invalid.rs:42:5
   |
42 |     #[try_stream(ok = (), error = ())] //~ ERROR may not be used at the same time
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: duplicate #[stream] attribute
  --> tests/ui/invalid.rs:48:13
   |
48 |             #[stream] //~ ERROR duplicate #[stream] attribute
   |             ^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> tests/ui/invalid.rs:56:13
   |
56 |             #[try_stream] //~ ERROR duplicate #[try_stream] attribute
   |             ^^^^^^^^^^^^^

error: duplicate #[for_await] attribute
  --> tests/ui/invalid.rs:63:9
   |
63 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
   |         ^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
  --> tests/ui/invalid.rs:70:13
   |
70 |             #[try_stream] //~ ERROR may not be used at the same time
   |             ^^^^^^^^^^^^^

error: duplicate #[stream] attribute
  --> tests/ui/invalid.rs:79:13
   |
79 |             #[stream] //~ ERROR duplicate #[stream] attribute
   |             ^^^^^^^^^

error: duplicate #[try_stream] attribute
  --> tests/ui/invalid.rs:88:13
   |
88 |             #[try_stream] //~ ERROR duplicate #[try_stream] attribute
   |             ^^^^^^^^^^^^^

error: duplicate #[for_await] attribute
  --> tests/ui/invalid.rs:96:9
   |
96 |         #[for_await] //~ ERROR duplicate #[for_await] attribute
   |         ^^^^^^^^^^^^

error: #[stream] and #[try_stream] may not be used at the same time
   --> tests/ui/invalid.rs:104:13
    |
104 |             #[try_stream] //~ ERROR may not be used at the same time
    |             ^^^^^^^^^^^^^

error: #[stream] attribute may only be used on async functions or async blocks
   --> tests/ui/invalid.rs:113:5
    |
113 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: #[stream] attribute may only be used on async functions or async blocks
   --> tests/ui/invalid.rs:116:5
    |
116 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: #[stream] attribute may only be used on async functions or async blocks
   --> tests/ui/invalid.rs:119:5
    |
119 |     #[stream(item = ())] //~ ERROR #[stream] attribute may only be used on async functions or async blocks
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)
