error[E0727]: `async` coroutines are not yet supported
  --> tests/ui/nested.rs:19:13
   |
19 |             yield i * i; //~ ERROR `async` coroutines are not yet supported [E0727]
   |             ^^^^^^^^^^^

error[E0308]: mismatched types
  --> tests/ui/nested.rs:19:19
   |
19 |             yield i * i; //~ ERROR `async` coroutines are not yet supported [E0727]
   |                   ^^^^^ expected `()`, found `i32`
