error[E0412]: cannot find type `L` in this scope
 --> tests/ui/unresolved-type.rs:5:17
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  |                 ^ not found in this scope
  |
help: you might be missing a type parameter
  |
6 | async fn foo<L>() {}
  |             +++

error[E0698]: type inside generator must be known in this context
 --> tests/ui/unresolved-type.rs:5:1
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  | ^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the enum `Poll`
  |
note: the type is part of the generator because of this `yield`
 --> tests/ui/unresolved-type.rs:5:1
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  | ^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)
